avy.el

```diff
@@ -465,15 +465,19 @@ multiple DISPLAY-FN invocations."
         (dolist (x avy--leafs)
           (funcall display-fn (car x) (cdr x))))
       (let ((char (funcall avy-translate-char-function (read-key)))
+	    window
             branch)
         (funcall cleanup-fn)
-        (if (setq branch (assoc char tree))
-            (progn
-              (setq avy-current-path
-                    (concat avy-current-path (string (avy--key-to-char char))))
-              (when (eq (car (setq tree (cdr branch))) 'leaf)
-                (throw 'done (cdr tree))))
-          (funcall avy-handler-function char))))))
+	(if (setq window (avy-mouse-event-window char))
```

Comment:
Like I said before, this change isn't needed: it's fine to let `avy-handler-function` get called here. It should handle the mouse event.

Reply:
​I had written two reasons why this was necessary to your personal email
today but I don't see a copy in my outbox now, so it looks like you did not
get that.
Here's a shorter version:

1. If the char sent is a window and it is not handled here, then the following call to (string (avy--key-to-char char)) will fail.  avy-current-path must be set here so that avy package users can reference it and have the full path saved there, e.g. for ace-window, any typed window label would be in there (whereas with the current code, it is always a character behind, which in the case of single character labels, it contains nothing).

2. avy-handler-function can be overridden by each package which means they would have to each duplicate this logic if moved to avy-handler-function. It is much more likely they will get it wrong at some point.

Can you point to any use-case where this causes any problem or fails to solve the problems it aims to (returning a clicked upon window and storing the selected window's label)?